engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsAnyNode.java

```diff
@@ -0,0 +1,609 @@
+              : selfWithWarnings;
+      Object other =
+          otherWarnLib.hasWarnings(otherWithWarnings) ? otherWarnLib.removeWarnings(otherWithWarnings)
+              : otherWithWarnings;
+      return equalsNode.execute(self, other);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  /** Interop libraries **/
+
+  @Specialization(guards = {
+      "selfInterop.isNull(selfNull)",
+      "otherInterop.isNull(otherNull)"
+  }, limit = "3")
+  boolean equalsNull(
+      Object selfNull, Object otherNull,
+      @CachedLibrary("selfNull") InteropLibrary selfInterop,
+      @CachedLibrary("otherNull") InteropLibrary otherInterop
+  ) {
+    return true;
+  }
+
+  @Specialization(guards = {
+      "selfInterop.isBoolean(selfBoolean)",
+      "otherInterop.isBoolean(otherBoolean)"
+  }, limit = "3")
+  boolean equalsBooleanInterop(
+      Object selfBoolean,
+      Object otherBoolean,
+      @CachedLibrary("selfBoolean") InteropLibrary selfInterop,
+      @CachedLibrary("otherBoolean") InteropLibrary otherInterop
+  ) {
+    try {
+      return selfInterop.asBoolean(selfBoolean) == otherInterop.asBoolean(otherBoolean);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @Specialization(guards = {
+      "selfInterop.isMetaObject(selfMeta)",
+      "otherInterop.isMetaObject(otherMeta)",
+  }, limit = "3")
+  boolean equalsMetaObjects(Object selfMeta, Object otherMeta,
+      @CachedLibrary("selfMeta") InteropLibrary selfInterop,
+      @CachedLibrary("otherMeta") InteropLibrary otherInterop,
+      @Cached EqualsAnyNode equalsNode) {
+    try {
+      Object selfMetaName = selfInterop.getMetaQualifiedName(selfMeta);
+      Object otherMetaName = otherInterop.getMetaQualifiedName(otherMeta);
+      return equalsNode.execute(selfMetaName, otherMetaName);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @Specialization(guards = {
+      "selfInterop.isDate(selfDateTime)",
+      "selfInterop.isTime(selfDateTime)",
+      "otherInterop.isDate(otherDateTime)",
+      "otherInterop.isTime(otherDateTime)"
+  }, limit = "3")
+  boolean equalsDateTimes(Object selfDateTime, Object otherDateTime,
+      @CachedLibrary("selfDateTime") InteropLibrary selfInterop,
+      @CachedLibrary("otherDateTime") InteropLibrary otherInterop) {
+    LocalDateTime selfJavaDateTime;
+    LocalDateTime otherJavaDateTime;
+    try {
+      selfJavaDateTime = LocalDateTime.of(
+          selfInterop.asDate(selfDateTime),
+          selfInterop.asTime(selfDateTime)
+      );
+      otherJavaDateTime = LocalDateTime.of(
+          otherInterop.asDate(otherDateTime),
+          otherInterop.asTime(otherDateTime)
+      );
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+    return selfJavaDateTime.isEqual(otherJavaDateTime);
+  }
+
```

Comment:
Our `EnsoDateTime` also contains a timezone. This comparison ignores it.

Do I understand correctly that `2022-12-20 13:16 CET == 2022-12-20 13:16 PDT` with this approach? That would not be good. Can we add the timezone to this comparison?